osf/management/commands/data_storage_usage.py

```diff
@@ -0,0 +1,635 @@
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_QUICK_FILE_SIZE_SUM_SQL,
+            content_type=abstractnode_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+
+        logger.debug('Gathering supplement summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_PREPRINT_SUPPLEMENT_SIZE_SUM_SQL,
+            content_type=abstractnode_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+        logger.debug('Gathering deleted file summary at {}'.format(datetime.datetime.now()))
+        cursor.execute(
+            DELETED_FILE_SIZE_SUM_SQL,
+            [
+                start,
+                end,
+            ]
+        )
+        summary_data = combine_summary_data(summary_data, cursor.fetchall())
+        logger.debug('Gathering total file summary at {}'.format(datetime.datetime.now()))
+        cursor.execute(
+            TOTAL_FILE_SIZE_SUM_SQL,
+            [
+                start,
+                end,
+            ]
+        )
+        summary_data = combine_summary_data(summary_data, cursor.fetchall())
+
+        logger.debug('Gathering preprint usage at {}'.format(datetime.datetime.now()))
+        filename = './data-usage-raw-preprints-{}-{}.csv'.format(start, end)
+
+        cursor.execute(
+            PREPRINT_LIST_SQL,
+            [
+                preprint_content_type,
+                preprint_content_type,
+                start,
+                end,
+            ]
+        )
+        if not dry_run:
+            logger.debug('Writing {} to zip.'.format(filename))
+            write_raw_data(cursor=cursor, zip_file=zip_file, filename=filename)
+
+        logger.debug('Gathering preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=ND_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+        logger.debug('Gathering regional preprint summary at {}'.format(datetime.datetime.now()))
+        summary_data = combine_summary_data(summary_data, summarize(
+            sql=REGIONAL_PREPRINT_SIZE_SUM_SQL,
+            content_type=preprint_content_type,
+            start=start,
+            end=end,
+            cursor=cursor,
+        ))
+
+    return summary_data
+
+
+def write_summary_data(filename, summary_data, remote_base_folder):
+    header_row = summary_data.keys()
+    summary_row = summary_data.values()
+    file_path = '{}{}'.format(TEMP_FOLDER, filename)
+    old_remote = requests.get(
+        url=remote_base_folder['files'],
+        auth=bearer_token_auth(DS_METRICS_OSF_TOKEN),
+        params={'filter[name]': filename},
+    ).json()
+    try:
+        if old_remote['meta']['total'] > 1:
```

Comment:
Should be `old_remote['links']['meta']['total']`

Reply:
Yeah, I'm looking at this now. Isn't that the old format? The `meta` on the response says version=2.0, not 2.14, and I'm trying to get it to be 2.14 because when I did this with the browsable api, I got:
```
    "meta": {
        "total": 1,
        "per_page": 10,
        "version": "2.14"
    },
    "links": {
        "self": "https://api.osf.io/v2/nodes/ezcuj/files/osfstorage/",
        "first": null,
        "last": null,
        "prev": null,
        "next": null
    }
```
And I'd like to be vaguely up to date with my api response. But I'm having trouble using the accept-header to change the version, and I'm about to try the query parameter. But if one of my assumptions is wrong, then that would make more sense.

Reply:
Query parameter worked. I wonder if our dev docs are misleading and it should be the `accept` header and not the `accept-header` as written. That's probably it.

Reply:
ah, I didn't realize we'd moved this into meta with later versions!  yes, I just ran and got a KeyError, so we might need to figure out how to request later versions, query param should work.